src/mono/mono/mini/monovm.c

```diff
@@ -184,6 +184,9 @@ parse_properties (int propertyCount, const char **propertyKeys, const char **pro
 		} else if (prop_len == 27 && !strncmp (propertyKeys [i], "System.Globalization.UseNls", 27)) {
 			// TODO: Ideally we should propagate this through AppContext options
 			g_setenv ("DOTNET_SYSTEM_GLOBALIZATION_USENLS", propertyValues [i], TRUE);
+		} else if (prop_len == 32 && !strncmp (propertyKeys [i], "System.Globalization.AppLocalIcu", 32)) {
+			// TODO: Ideally we should propagate this through AppContext options
+			g_setenv ("DOTNET_SYSTEM_GLOBALIZATION_APPLOCALICU", propertyValues [i], TRUE);
```

Comment:
I assume the value validation is not needed here. right?

Reply:
You mean that the property value is in the right/expected form?

Reply:
I mean if someone set some bogus values in the `propertyValues[i]`  is not going to cause any problem when calling `g_setenv` and this value is going to be validated later.

Reply:
That I'm not sure... so long the value is set as an env variable correctly it will be validated when loading ICU. @CoffeeFlux do you know?

Reply:
It seems like we shouldn't have any problems with this, it just uses the os APIs directly and sets whatever string is passed down: 
https://github.com/dotnet/runtime/blob/cd622cd50c7104fc14a25fb199f8377472920f2e/src/mono/mono/eglib/gmisc-unix.c#L94
https://github.com/dotnet/runtime/blob/cd622cd50c7104fc14a25fb199f8377472920f2e/src/mono/mono/eglib/gmisc-win32.c#L85

Reply:
This should be fine. The environmental variables are really just a short-term hack around the fact we don't have proper helper functions for manipulating AppContext from the native side., and they're just read in and compared as strings in a couple places in managed code. I'll open an issue about that though—we should really do this properly instead of relying on this hack for much longer.

Reply:
Thanks @CoffeeFlux, I agree with that.